test/model/src/driver/debug_log_parser.cpp

```diff
@@ -0,0 +1,503 @@
+    uint64_t id = fileid & (WT_LOGOP_IGNORE - 1);
+    auto table_itr = _fileid_to_table.find(id);
+    if (table_itr == _fileid_to_table.end())
+        throw model_exception("Unknown file ID: " + std::to_string(id));
+    return table_itr->second;
+}
+
+/*
+ * debug_log_parser::metadata_apply --
+ *     Apply the given operation to the model.
+ */
+void
+debug_log_parser::metadata_apply(const row_put &op)
+{
+    std::string key =
+      std::get<std::string>(data_value::unpack(op.key.c_str(), op.key.length(), "S"));
+    std::string value =
+      std::get<std::string>(data_value::unpack(op.value.c_str(), op.value.length(), "S"));
+
+    /* Parse the configuration string. */
+    std::shared_ptr<config_map> m =
+      std::make_shared<config_map>(config_map::from_string(value.c_str()));
+
+    /* Remember the metadata. */
+    _metadata[key] = m;
+
+    /* Special handling for column groups. */
+    if (starts_with(key, "colgroup:")) {
+        std::string name = key.substr(std::strlen("colgroup:"));
+        if (name.find(':') != std::string::npos)
+            throw model_exception("The model does not currently support column groups");
+
+        std::string source = m->get_string("source");
+        _file_to_colgroup_name[source] = name;
+
+        /* Establish mapping from the file ID to the table name and table object, if possible. */
+        auto i = _file_to_fileid.find(source);
+        if (i != _file_to_fileid.end()) {
+            _fileid_to_table_name[i->second] = name;
+            if (!_database.contains_table(name))
+                throw model_exception("The database does not yet contain table " + name);
+            _fileid_to_table[i->second] = _database.table(name);
+        }
+    }
+
+    /* Special handling for files. */
+    if (starts_with(key, "file:")) {
+        uint64_t id = m->get_uint64("id");
+        _fileid_to_file[id] = key;
+        _file_to_fileid[key] = id;
+
+        /* Establish mapping from the file ID to the table name and table object, if possible. */
+        auto i = _file_to_colgroup_name.find(key);
+        if (i != _file_to_colgroup_name.end()) {
+            _fileid_to_table_name[id] = i->second;
+            if (!_database.contains_table(i->second))
+                throw model_exception("The database does not yet contain table " + i->second);
+            _fileid_to_table[id] = _database.table(i->second);
+        }
+    }
+
+    /* Special handling for LSM. */
+    if (starts_with(key, "lsm:"))
+        throw model_exception("The model does not currently support LSM");
+
+    /* Special handling for tables. */
+    if (starts_with(key, "table:")) {
+        std::string name = key.substr(std::strlen("table:"));
+        if (!_database.contains_table(name)) {
+            kv_table_ptr table = _database.create_table(name);
+            table->set_key_value_format(m->get_string("key_format"), m->get_string("value_format"));
+        }
+    }
```

Comment:
Good point—it's probably a good idea to raise an error if there is some surprising metadata item instead of silently ignoring it, just in case this would matter for the verification correctness. I don't think there are metadata items without a valid prefix? Anyways, I refactored it so that it would be also included in the catch-all clause that throws an exception.